src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,81 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 1 << 4;
+    static final float LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private float threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = DEFAULT_CAPACITY * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int index = getHashIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (key == node.key || key != null && key.equals(node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        node = new Node<K, V>(key, value, table[index]);
+        table[index] = node;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getHashIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (key == node.key || key != null && key.equals(node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length << 1];
+        threshold = table.length * LOAD_FACTOR;
```

Comment:
```suggestion
        threshold = newTable.length * LOAD_FACTOR;
```
У teble пока ещё старая длинна

Reply:
I lost it💁🏼. Changed.